enterprise_subsidy/apps/subsidy/rules.py

```diff
@@ -0,0 +1,140 @@
+"""
+Rules needed to restrict access to the enterprise subsidy service.
+"""
+import crum
+import rules
+from edx_rbac.utils import request_user_has_implicit_access_via_jwt, user_has_access_via_database
+from edx_rest_framework_extensions.auth.jwt.authentication import get_decoded_jwt_from_auth
+from edx_rest_framework_extensions.auth.jwt.cookies import get_decoded_jwt
+
+from enterprise_subsidy.apps.subsidy.constants import (
+    ENTERPRISE_SUBSIDY_ADMIN_ROLE,
+    ENTERPRISE_SUBSIDY_LEARNER_ROLE,
+    ENTERPRISE_SUBSIDY_OPERATOR_ROLE,
+    PERMISSION_CAN_CREATE_TRANSACTIONS,
+    PERMISSION_CAN_READ_SUBSIDIES,
+    PERMISSION_CAN_READ_TRANSACTIONS
+)
+from enterprise_subsidy.apps.subsidy.models import EnterpriseSubsidyRoleAssignment
+
+
+@rules.predicate
+def has_implicit_access_to_subsidy_operator(user, context):  # pylint: disable=unused-argument
+    """
+    Check that the requesting user has implicit openedx operator access to the given context.
+
+    Returns:
+        bool: True if the user has access.
+    """
+    if not context:
+        return False
+    request = crum.get_current_request()
+    decoded_jwt = get_decoded_jwt(request) or get_decoded_jwt_from_auth(request)
+    return request_user_has_implicit_access_via_jwt(decoded_jwt, ENTERPRISE_SUBSIDY_OPERATOR_ROLE, context)
+
+
+@rules.predicate
+def has_explicit_access_to_subsidy_operator(user, context):
+    """
+    Check that the given user has explicit openedx operator access to the given context.
+
+    Returns:
+        bool: True if the user has access.
+    """
+    if not context:
+        return False
+    return user_has_access_via_database(
+        user,
+        ENTERPRISE_SUBSIDY_OPERATOR_ROLE,
+        EnterpriseSubsidyRoleAssignment,
+        context,
+    )
```

Comment:
Nit: it's never struck me before how similar the implicit/explicit bodies of all our rules predicates are.  Might be worth adding a little private helper e.g.
```python
def _user_has_implicit_access_via_role(user, context, role):
    if not context:
        return False
    request = crum.get_current_request()
    decoded_jwt = get_decoded_jwt(request) or get_decoded_jwt_from_auth(request)
    return request_user_has_implicit_access_via_jwt(decoded_jwt, role, context)
```
Longer-term, we should add these helpers to `edx_rbac.utils`